File: Java/Strategy/src/duck/MallardDuck.java
package duck;

import behavior.FlyWithWings;
import behavior.Quack;

public class MallardDuck extends Duck{
    public MallardDuck() {
        TAG = "MallardDuck";
        quackBehavior = new Quack();
        flyBehavior = new FlyWithWings();
    }
}

File: Kotlin/src/main/kotlin/chapter1_strategy/MallardDuck.kt
package chapter1

class MallardDuck : Duck() {
    override var flyBehavior: FlyBehavior = FlyWithWings()
    override var quackBehavior: QuackBehavior = Quack()

    override fun display() {
        println("저는 물오리입니다")
    }
}